### PR TITLE
JP-2626: Add check to all lv3_asns to prevent single-obs group candidates

### PR DESCRIPTION
**Pull request by tapastro**
Closes #
Resolves [JP-2626](https://jira.stsci.edu/browse/JP-2626)

**Description**

This PR adds a finalization check to level 3 association candidates that prevents the creation of any `group` association candidate with members from only one group.

Checklist
- [ ] Tests
- [ ] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

**Comment by tapastro** [comment]
Latest commit to fix last regtest error from Jenkins runs, should be in a workable state now.

File: jwst/associations/lib/rules_level3_base.py [hunk 1]
```diff
@@ -478,6 +478,21 @@ def ok_candidate(self, member=None):
         """
         return self.acid.type.lower() not in INVALID_AC_TYPES
 
+    def finalize(self):
+        if self.acid.type.lower() == 'group':
+            if len(self.current_product['members']):
+                try:
+                    # Janky extraction of obs_num from filenames
```
Comment (hbushouse) [inline jwst/associations/lib/rules_level3_base.py:485]: "Janky"  Is that a technical term?

**Review by hbushouse — APPROVED** [review]
Looks OK to me, but I'm no asn rule expert.

**Comment by hbushouse** [comment]
py3.10 CI failure is just another connection problem retrieving files from crds.

File: jwst/associations/lib/rules_level3_base.py [hunk 2]
```diff
@@ -478,6 +478,21 @@ def ok_candidate(self, member=None):
         """
         return self.acid.type.lower() not in INVALID_AC_TYPES
 
+    def finalize(self):
+        if self.acid.type.lower() == 'group':
```
Comment (stscieisenhamer) [inline jwst/associations/lib/rules_level3_base.py:482]: This is possibly too restrictive? In fact, I wonder if it should be the other way: `acid.type != 'observation'`. All other candidate types should be multi-observation, though coronagraphic/ami should be reviewed. There is a possibility that those modes could be still valid for single observations.

File: jwst/associations/lib/rules_level3_base.py (hunk 1 above)
Comment (stscieisenhamer) [reply jwst/associations/lib/rules_level3_base.py:485]: I like "janky".

However, there is a more "association-like" way of doing this. There is an "informational" constraint, called `Constraint_Obsnum`, that is part of the `AsnMixin_Science` class. The whole point of this constraint is to just gather all the observation numbers that are part of the association.

The multi-observation check can then be to ensure that the list of `self.constraints.['obs_num'].found_values` is greater than one.

Comment (stscieisenhamer) [reply jwst/associations/lib/rules_level3_base.py:485]: This will take away the filename dependency, and make it contolled by the input association pool.

**Review by stscieisenhamer — CHANGES_REQUESTED** [review]

File: jwst/associations/lib/rules_level3_base.py (hunk 1 above)
Comment (stscieisenhamer) [reply jwst/associations/lib/rules_level3_base.py:485]: Some further thoughts:
- This method should be moved to `Asn_MixinScience`
- Ensure that parent class `finalize` is also called, adding the `super` call at the end of the method.

**Pull request merged**